backend/app/usecase/shortlink/updater_test.go

```diff
@@ -149,6 +135,96 @@ func TestShortLinkUpdaterPersist_UpdateShortLink(t *testing.T) {
 			},
 			expectedHasErr: true,
 		},
+		{
+			name:  "alias is empty",
+			alias: "git",
+			shortlinks: shortLinks{
+				"git": entity.ShortLink{
+					Alias:     "git",
+					LongLink:  "https://github.com/short-d",
+					UpdatedAt: &now,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "gopher@golang.org",
+			},
+			updateArgs: entity.ShortLinkInput{
+				CustomAlias: ptr.String(""),
+			},
+			relationUsers: []entity.User{
+				{ID: "1"},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:     "git",
+					LongLink:  "https://github.com/short-d",
+					UpdatedAt: &now,
+				},
+			},
+			expectedHasErr: true,
+		},
+		{
+			name:  "long link not changed if not specified",
```

Comment:
This test currently fails when it should pass. Leaving a review comment for myself to investigate why the user relation check on line 396 of `updater_test.go` fails when this short link is updated.

Reply:
After some investigation, this is failing because the fake short link repository does not update the user_short_link relation table. The SQL version implicitly updates the user_short_link relation table via `ON UPDATE CASCADE` trigger. I will think of a way to implement this cleanly using the current fake repo implementation, or I'll make a new issue and move this test there.

Reply:
Let's pass in fake short link repo as a dependency of `user_short_link` and add a helper method to update the alias for now. And add a TODO to switch to eventbus for change propagation across multiple repositories.